testing/src/main/java/io/grpc/internal/testing/CensusTestUtils.java

```diff
@@ -0,0 +1,210 @@
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.internal.testing;
+
+import com.google.census.CensusContext;
+import com.google.census.CensusContextFactory;
+import com.google.census.Metric;
+import com.google.census.MetricMap;
+import com.google.census.MetricName;
+import com.google.census.TagKey;
+import com.google.census.TagValue;
+import com.google.common.collect.ImmutableMap;
+
+import io.grpc.Context;
+import io.grpc.internal.GrpcUtil;
+import io.grpc.internal.NoopCensusContextFactory;
+import io.grpc.internal.StatsTraceContext;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class CensusTestUtils {
+  private CensusTestUtils() {
+  }
+
+  /**
+   * Creates a StatsTraceContext that does nothing.
+   */
+  public static StatsTraceContext noopStatsTraceContext() {
+    return StatsTraceContext.createOnClient(
+        "noopservice/noopmethod", NoopCensusContextFactory.INSTANCE,
+        GrpcUtil.STOPWATCH_SUPPLIER);
+  }
+
+  public static class MetricsRecord {
+    public final ImmutableMap<TagKey, TagValue> tags;
+    public final MetricMap metrics;
+
+    private MetricsRecord(ImmutableMap<TagKey, TagValue> tags, MetricMap metrics) {
+      this.tags = tags;
+      this.metrics = metrics;
+    }
+
+    public Double getMetric(MetricName metricName) {
+      for (Metric m : metrics) {
+        if (m.getName().equals(metricName)) {
+          return m.getValue();
+        }
+      }
+      return null;
+    }
+
+    public long getMetricAsLongOrFail(MetricName metricName) {
```

Comment:
heh it will fail :) with NPE. there's some logic here that could use a comment, too